src/components/vm/overview/vsock.tsx

```diff
@@ -143,31 +143,36 @@ export const VsockModal = ({
 
     const Dialogs = useDialogs();
 
-    const save = () => {
+    const save = async () => {
+        const values = {
+            cid: {
+                auto: auto ? "yes" : "no",
+                address,
+            }
+        };
+
         setActionInProgress("save");
-        return domainSetVsock({
-            connectionName: vm.connectionName,
-            vmName: vm.name,
-            hotplug: vm.state === "running",
-            permanent: vm.persistent,
-            auto: auto ? "yes" : "no",
-            address: String(address),
-            isVsockAttached,
-        })
-                .then(Dialogs.close, exc => setDialogError({ text: _("Failed to configure vsock"), detail: exc.message }))
-                .finally(() => setActionInProgress(undefined));
+        try {
+            if (isVsockAttached)
+                await virtXmlEdit(vm, "vsock", 1, values);
+            else
+                await virtXmlHotAdd(vm, "vsock", values);
+            Dialogs.close();
+        } catch (exc) {
+            setDialogError({ text: _("Failed to configure vsock"), detail: String(exc) });
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2334-68be91e9-20250929-092324-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)